Lib/_pyio.py

```diff
@@ -324,6 +324,8 @@ class OpenWrapper:
     __doc__ = DocDescriptor()
 
     def __new__(cls, *args, **kwargs):
+        if len(args) >= 2 and "b" not in args[1]:
+            kwargs["encoding"] = text_encoding(kwargs.get("encoding"))
```

Comment:
Why does _io emit the warning in _io.TextIOWrapper, but you chose to modify OpenWrapper in _pyio?

You might extract this fix into a separated PR.

Reply:
Because EncodingWarning was emitted for the OpenWrapper, not for the caller.

Reply:
There was a bug in this `if` statement. I revert this.